WordPress/src/main/java/org/wordpress/android/ui/mlp/ModalLayoutPickerFragment.kt

```diff
@@ -0,0 +1,143 @@
+package org.wordpress.android.ui.mlp
+
+import android.content.Context
+import android.os.Bundle
+import android.view.KeyEvent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.WindowManager
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProviders
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import kotlinx.android.synthetic.main.modal_layout_picker_bottom_toolbar.*
+import kotlinx.android.synthetic.main.modal_layout_picker_fragment.*
+import kotlinx.android.synthetic.main.modal_layout_picker_header.*
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.util.WPActivityUtils
+import org.wordpress.android.util.setVisible
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.mlp.ModalLayoutPickerViewModel
+import javax.inject.Inject
+
+/**
+ * Implements the Modal Layout Picker UI based on the [BottomSheetDialogFragment] to inherit the container behavior
+ */
+class ModalLayoutPickerFragment : BottomSheetDialogFragment() {
```

Comment:
Just as a general naming suggestion but not requiring a change here. `ModalLayoutPickerFragment` feels like it's really inforcing the display mechanism. Because of that on iOS, I went with `GutenbergLayoutPicker__` because that felt like it described more of what it does instead of how it should be displayed plus what it does.

Reply:
I agree with you on this Chip. Since the Modal/MLP naming has infected a lot of my classes an resource naming I will leave this refactoring last to avoid any conflicts